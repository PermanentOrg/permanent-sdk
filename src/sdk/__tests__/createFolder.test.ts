import path from "node:path";
import nock from "nock";
import { createFolder } from "..";

describe("createFolder", () => {
	it("should create a folder", async () => {
		nock("https://permanent.local")
			.post("/api/folder/post", {
				displayName: "testFolder",
				parentFolderId: 1,
				failOnDuplicateName: false,
			})
			.replyWithFile(
				200,
				path.join(__dirname, "fixtures", "createFolder", "folder.json"),
				{
					"Content-Type": "application/json",
				},
			);

		const folder = await createFolder(
			{
				bearerToken: "12345",
				baseUrl: "https://permanent.local/api",
			},
			{
				folder: {
					name: "testFolder",
				},
				parentFolder: {
					id: 1,
				},
			},
		);

		expect(folder).toMatchSnapshot();
	});
	it("should use the failOnDuplicateName parameter if set", async () => {
		nock("https://permanent.local")
			.post("/api/folder/post", {
				displayName: "testFolder",
				parentFolderId: 1,
				failOnDuplicateName: true,
			})
			.replyWithFile(
				200,
				path.join(__dirname, "fixtures", "createFolder", "folder.json"),
				{
					"Content-Type": "application/json",
				},
			);

		const folder = await createFolder(
			{
				bearerToken: "12345",
				baseUrl: "https://permanent.local/api",
			},
			{
				folder: {
					name: "testFolder",
				},
				parentFolder: {
					id: 1,
				},
				failOnDuplicateName: true,
			},
		);

		expect(folder).toMatchSnapshot();
	});
});
