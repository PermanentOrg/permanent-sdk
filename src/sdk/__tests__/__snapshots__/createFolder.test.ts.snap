// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createFolder should create a folder 1`] = `
{
  "archiveRecords": [],
  "createdAt": 2022-12-15T17:04:32.000Z,
  "displayDate": 2022-12-15T17:04:31.000Z,
  "fileSystemCompatibleName": "testFolder (2)",
  "fileSystemId": 7,
  "folders": [],
  "id": 7,
  "name": "testFolder",
  "size": 0,
  "updatedAt": 2022-12-15T17:04:32.000Z,
}
`;
